src/key.h

```diff
@@ -37,6 +38,7 @@ class CKey
      */
     static const unsigned int SIZE            = 279;
     static const unsigned int COMPRESSED_SIZE = 214;
+    static constexpr unsigned int KEY_SIZE = 32;
```

Comment:
Seems slightly odd to use a `vector` when we want something that's always 32 bytes. Could perhaps have `secure_unique_ptr<std::array<unsigned char, KEY_SIZE>> keydata` instead, essentially replacing `.resize(32)` with a `make_secure_unique()` call and replacing `ClearShrink` with `keydata = nullptr`.

```c++
template<typename T>
struct SecureUniqueDeleter {
    void operator()(T* t) noexcept {
        secure_allocator<T>().deallocate(t, 1);
    }
};

template<typename T>
using secure_unique_ptr = std::unique_ptr<T,SecureUniqueDeleter<T>>;

template<typename T, typename... Args>
secure_unique_ptr<T> make_secure_unique(Args&&... as)
{
    T* p = secure_allocator<T>().allocate(1);

    // initialize in place, and return as secure_unique_ptr
    return secure_unique_ptr<T>(new (p) T(std::forward(as)...));
}

...
using KeyType = std::array<unsigned char, 32>;
secure_unique_ptr<KeyType> keydata{nullptr};
```

Doing this instead of a vector requires replacing `keydata.data()` with `keydata->data()` (etc), and writing an explicit copy constructor for `CKey`. This would save 16 bytes per `CKey` (vector presumably stores a 64 bit size/capacity), but it's not in the secure allocator pool, so we probably don't care.